__tests__/unit/kafkaConsumer.js

```diff
@@ -11,52 +11,27 @@ const kafka = require('no-kafka');
 jest.mock('no-kafka');
 
 describe('test/unit/consumer.js', () => {
-  it('TopicHandlers gives you an object with mapping from topicName and throws no error',
-    async () => {
-    const { topicHandlers } = await kafkaConsumer.initConsumer();
-    expect(topicHandlers.foo).toBeInstanceOf(Function);
-    expect(topicHandlers.bar).toBeInstanceOf(Function);
-  });
-
   it('Creates the consumer with the right arguments', async () => {
-    const simpleConsumerMock = jest.spyOn(kafka, 'SimpleConsumer');
+    const GroupConsumerMock = jest.spyOn(kafka, 'GroupConsumer');
     const topicHandlers = await kafkaConsumer.initConsumer();
-    expect(simpleConsumerMock).toHaveBeenCalledWith({
-      clientId: `consumer-${process.pid}`,
-      connectionString: 'test-url',
-      ssl: {
-        cert: 'test-cert',
-        key: 'test-key',
-      },
-      maxWaitTime: 100,
-      maxBytes: (1024 * 1024),
-      idleTimeout: 1000,
-    });
-  });
-
-  it('subscribe throws an error', async () => {
-    const simpleConsumerMock = jest.spyOn(kafka, 'SimpleConsumer');
-    simpleConsumerMock.mockImplementationOnce(() => ({
-        subscribe: () => {
-          throw new Error();
-        },
-
-        init: () => {
+    expect(GroupConsumerMock).toHaveBeenCalledWith(
```

Comment:
Any way to test the actual group functionality?

Reply:
Unfortunately, we can't without setting full integration test (Since the GroupConsumer is basically an API). (So far we don't have a way to set up a Kafka as an add-on on heroku for testing).

Reply:
Would this help? https://devcenter.heroku.com/articles/kafka-on-heroku

Reply:
This is actually different